core/internal/gethwrappers/abigen.go

```diff
@@ -54,4 +66,239 @@ func Abigen(a AbigenArgs) {
+	contractName := getContractName(fileNode)
+	fileNode = addAddressField(contractName, fileNode)
+	fileNode = replaceAnonymousStructs(contractName, fileNode)
+	bs = generateCode(fset, fileNode)
+	bs = writeAdditionalMethods(contractName, bs)
+	fset, fileNode = parseFile(bs)
+	fileNode = writeInterface(contractName, fileNode)
+	bs = generateCode(fset, fileNode)
+	bs = addHeader(bs)
+
+	err = ioutil.WriteFile(path, bs, 0600)
+	if err != nil {
+		Exit("Error while writing improved abigen source", err)
+	}
+}
+
+func parseFile(bs []byte) (*token.FileSet, *ast.File) {
+	fset := token.NewFileSet()
+	fileNode, err := parser.ParseFile(fset, "", string(bs), parser.AllErrors)
+	if err != nil {
+		Exit("Error while improving abigen output", err)
+	}
+	return fset, fileNode
+}
+
+func generateCode(fset *token.FileSet, fileNode *ast.File) []byte {
+	var buf bytes.Buffer
+	err := format.Node(&buf, fset, fileNode)
+	if err != nil {
+		Exit("Error while writing improved abigen source", err)
+	}
+	return buf.Bytes()
+}
+
+func getContractName(fileNode *ast.File) string {
+	// Grab the contract name. It's always the first type in the file.
+	var contractName string
+	astutil.Apply(fileNode, func(cursor *astutil.Cursor) bool {
+		x, is := cursor.Node().(*ast.TypeSpec)
+		if !is {
+			return true
+		}
+		if contractName == "" {
+			contractName = x.Name.Name
+		}
+		return false
+	}, nil)
+	return contractName
+}
+
+func addAddressField(contractName string, fileNode *ast.File) *ast.File {
+	// Add an exported `.Address` field to the contract struct
+	fileNode = astutil.Apply(fileNode, func(cursor *astutil.Cursor) bool {
+		x, is := cursor.Node().(*ast.StructType)
+		if !is {
+			return true
+		}
+		theType, is := cursor.Parent().(*ast.TypeSpec)
+		if !is {
+			return false
+		} else if theType.Name.Name != contractName {
+			return false
+		}
+
+		addrField := &ast.Field{
+			Names: []*ast.Ident{ast.NewIdent("address")},
+			Type: &ast.SelectorExpr{
+				X:   ast.NewIdent("common"),
+				Sel: ast.NewIdent("Address"),
+			},
+		}
+		x.Fields.List = append([]*ast.Field{addrField}, x.Fields.List...)
+
+		return false
+	}, nil).(*ast.File)
+
+	// Add the field to the return value of the constructor
+	fileNode = astutil.Apply(fileNode, func(cursor *astutil.Cursor) bool {
+		x, is := cursor.Node().(*ast.FuncDecl)
+		if !is {
+			return true
+		} else if x.Name.Name != "New"+contractName {
+			return false
+		}
+
+		for _, stmt := range x.Body.List {
+			returnStmt, is := stmt.(*ast.ReturnStmt)
+			if !is {
+				continue
+			}
+			lit, is := returnStmt.Results[0].(*ast.UnaryExpr).X.(*ast.CompositeLit)
+			if !is {
+				continue
+			}
+			kvExpr := &ast.KeyValueExpr{
+				Key:   ast.NewIdent("address"),
+				Value: ast.NewIdent("address"),
+			}
+			lit.Elts = append([]ast.Expr{kvExpr}, lit.Elts...)
+		}
+
+		return false
+	}, nil).(*ast.File)
+
+	return fileNode
+}
+
+func replaceAnonymousStructs(contractName string, fileNode *ast.File) *ast.File {
```

Comment:
Yeah they're used in the fluxmonitor, `OracleRoundState` and `LatestRoundData`. Alternative is to re-define the otherwise anonymous structs, which can be tedious to do whenever we want to use a new contract function, and also buggy (bug from last month was introduced because the struct types didn't match b/t abigen and our named type).